File: sdk/src/test/java/com/cxense/cxensesdk/ConversionEventConverterTest.java
package com.cxense.cxensesdk;

import com.cxense.cxensesdk.model.ConversionEvent;
import com.google.gson.Gson;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertNull;
import static org.junit.Assert.assertTrue;
import static org.mockito.Mockito.verify;
import static org.powermock.api.mockito.PowerMockito.mock;

public class ConversionEventConverterTest extends BaseTest {
    private Gson gson;
    private ConversionEventConverter converter;
    private ConversionEvent event;

    @Before
    public void setUp() throws Exception {
        super.setUp();
        gson = mock(Gson.class);
        converter = new ConversionEventConverter(gson);
        event = mock(ConversionEvent.class);
    }

    @Test
    public void canConvert() {
        assertTrue(converter.canConvert(event));
    }

    @Test
    public void toQueryMap() {
        assertNull(converter.toQueryMap(event));
    }

    @Test
    public void toEventRecord() throws Exception {
        assertNotNull(converter.toEventRecord(event));
        verify(gson).toJson(event);
    }
}

File: sdk/src/test/java/com/cxense/cxensesdk/ConversionEventConverterTest.kt
package com.cxense.cxensesdk

import com.cxense.cxensesdk.model.ConversionEvent
import com.nhaarman.mockitokotlin2.any
import com.nhaarman.mockitokotlin2.doReturn
import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.verify
import com.squareup.moshi.JsonAdapter
import kotlin.test.Test
import kotlin.test.assertFalse
import kotlin.test.assertNotNull
import kotlin.test.assertNull
import kotlin.test.assertTrue

class ConversionEventConverterTest {
    private val jsonAdapter: JsonAdapter<ConversionEvent> = mock {
        on { toJson(any()) } doReturn "{}"
    }
    private val converter = ConversionEventConverter(jsonAdapter)
    private val event: ConversionEvent = mock()

    @Test
    fun canConvertConversionEvent() {
        assertTrue {
            converter.canConvert(event)
        }
    }

    @Test
    fun canConvertOtherEvent() {
        assertFalse {
            converter.canConvert(mock())
        }
    }

    @Test
    fun toEventRecordConversionEvent() {
        assertNotNull(converter.toEventRecord(event))
        verify(jsonAdapter).toJson(any())
    }

    @Test
    fun toEventRecordOtherEvent() {
        assertNull(converter.toEventRecord(mock()))
    }
}
